src/canisters/frontend/ic-certified-assets/src/http.rs

```diff
@@ -80,6 +82,45 @@ impl HttpRequest {
             .iter()
             .find_map(|(k, v)| k.eq_ignore_ascii_case(header_key).then_some(v))
     }
+
+    pub fn redirect_from_raw_to_certified_domain(&self) -> HttpResponse {
+        #[cfg(not(test))]
+        let location = format!(
+            "https://{canister_id}.ic0.app{path}",
+            canister_id = ic_cdk::api::id().to_text(),
+            path = self.url
+        );
+        #[cfg(test)]
+        let location = format!(
+            "https://{canister_id}.ic0.app{path}",
+            canister_id = self.get_canister_id(),
+            path = self.url
+        );
+        HttpResponse::build_redirect(HTTP_REDIRECT_PERMANENT, location)
+    }
+
+    #[cfg(test)]
+    pub fn get_canister_id(&self) -> &str {
+        if let Some(host_header) = self.get_header_value("Host") {
+            if host_header.contains(".localhost") || host_header.contains(".app") {
+                return host_header.split('.').next().unwrap();
+            } else if let Some(t) = self.url.split("canisterId=").nth(1) {
+                let x = t.split_once('&');
+                if let Some(c) = x {
+                    return c.0;
+                }
+            }
+        }
+        "canister-id-is-unreachable"
```

Comment:
Since this is just for tests and not expected to be reached, maybe this should panic